core/client/fs/src/main/java/alluxio/client/block/policy/CapacityBaseRandomPolicy.java

```diff
@@ -31,18 +42,39 @@
 @ThreadSafe
 public class CapacityBaseRandomPolicy implements BlockLocationPolicy {
 
+  private final Cache<Long, List<WorkerNetAddress>> mBlockLocationCache;
+
+  private final int mMaxReplicaSize;
+
   /**
    * Constructs a new {@link CapacityBaseRandomPolicy}.
    *
    * @param conf Alluxio configuration
    */
   public CapacityBaseRandomPolicy(AlluxioConfiguration conf) {
+    mBlockLocationCache =
+        CacheBuilder.newBuilder().maximumSize(10000)
+            .expireAfterWrite(10, TimeUnit.MINUTES).build();
+    mMaxReplicaSize = conf.getInt(PropertyKey.USER_FILE_REPLICATION_MAX);
+  }
+
+  @VisibleForTesting
+  CapacityBaseRandomPolicy(int maxReplicaSize) {
```

Comment:
-> CapacityBaseRandomPolicy(AlluxioConfiguration conf, int maxReplicaSize)

and than `public CapacityBaseRandomPolicy(AlluxioConfiguration conf)` call this(conf, conf.getInt(PropertyKey.USER_FILE_REPLICATION_MAX))